src/query/sql/src/executor/physical_plans/physical_hash_join.rs

```diff
@@ -320,8 +320,28 @@ impl PhysicalPlanBuilder {
                 probe_fields.extend(build_fields);
                 probe_fields
             }
-            JoinType::LeftSemi | JoinType::LeftAnti => probe_fields,
-            JoinType::RightSemi | JoinType::RightAnti => build_fields,
+            JoinType::LeftSemi | JoinType::LeftAnti | JoinType::RightSemi | JoinType::RightAnti => {
+                let (result_fields, dropped_fields) = if join.join_type == JoinType::LeftSemi
+                    || join.join_type == JoinType::LeftAnti
+                {
+                    (probe_fields, build_fields)
+                } else {
+                    (build_fields, probe_fields)
+                };
+                for field in dropped_fields.iter() {
+                    if !result_fields.iter().any(|x| x.name() == field.name()) {
+                        match field.name().parse::<usize>() {
+                            Ok(index) if column_projections.contains(&index) => {
+                                return Err(ErrorCode::SemanticError(
```

Comment:
Can we pick a span and call `.set_span()` on the error?

Reply:
Currently `required(ColumnSet)` only contains column index, maybe we can add span to `required` later.